OCA-UFCG/oca-frontend
# Refactoring map page to make a better use of state control and context

## What this PR does?

I added a context that controls all the data that flows between components in the map page. With the context we don't need to share state through the components I just added a context that contains all this mutual data and the components that are under it's umbrella can access the state's data.

## How to test it?

This PR changes the code but hopefully doesn't add any bugs, so to test it I think that you can make the normal use of the map's page and section and see if it's working fine.

## Good material on context and useMemo

- [Context](https://react.dev/learn/passing-data-deeply-with-context#step-1-create-the-context);
- [useMemo](https://react.dev/reference/react/useMemo)
- [useMemo](https://www.freecodecamp.org/news/how-to-work-with-usememo-in-react/)
